packages/@headlessui-react/src/hooks/use-controllable.ts

```diff
@@ -4,10 +4,11 @@ import { useEvent } from './use-event'
 export function useControllable<T>(
   controlledValue: T | undefined,
   onChange?: (value: T) => void,
-  defaultValue?: T
+  defaultValue?: T,
+  isControlled?: boolean
 ) {
   let [internalValue, setInternalValue] = useState(defaultValue)
-  let isControlled = controlledValue !== undefined
+  isControlled = isControlled ?? controlledValue !== undefined
```

Comment:
I think this is fine, we _could_ apply this to other components as well but I wonder if that actually solves something or not.

I would say keep it like this somehow (unless there is an obvious bug), and if no obvious bug exists but later an issue is opened for this then we can add the missing `isControlled` boolean checks.

---

Thinking about this a bit more, I would honestly almost argue that the `value={undefined}` was already correct to determine whether or not the component was controlled vs uncontrolled, because this is also how React behaves: https://codesandbox.io/s/sleepy-snow-v4m8eu?file=/src/App.js

So I wonder if the better approach would be to mark the `value` as `null` in our Tailwind UI components (if that's not already the case; assuming that's where the error occurred initially?)

Reply:
Oh yeah that's a super good point. I wonder then if we can replicate the error message like React has instead to guide users to write the correct thing. I'll look around some more.